src/gui/guiImage.cpp

```diff
@@ -0,0 +1,201 @@
+			checkBounds(clippingRect);
+			clippingRect.clipAgainst(AbsoluteClippingRect);
+
+			driver->draw2DImage(Texture, AbsoluteRect.UpperLeftCorner, sourceRect,
+								&clippingRect, Color, UseAlphaChannel);
+		}
+	}
+	else if ( DrawBackground )
+	{
+		core::rect<s32> clippingRect(AbsoluteClippingRect);
+		checkBounds(clippingRect);
+
+		skin->draw2DRectangle(this, skin->getColor(EGDC_3D_DARK_SHADOW), AbsoluteRect, &clippingRect);
+	}
+
+	IGUIElement::draw();
+}
+
+
+//! sets if the image should use its alpha channel to draw itself
+void GUIImage::setUseAlphaChannel(bool use)
+{
+	UseAlphaChannel = use;
+}
+
+
+//! sets if the image should use its alpha channel to draw itself
+void GUIImage::setScaleImage(bool scale)
+{
+	ScaleImage = scale;
+}
+
+
+//! Returns true if the image is scaled to fit, false if not
+bool GUIImage::isImageScaled() const
+{
+	return ScaleImage;
+}
+
+//! Returns true if the image is using the alpha channel, false if not
+bool GUIImage::isAlphaChannelUsed() const
+{
+	return UseAlphaChannel;
+}
+
+//! Sets the source rectangle of the image. By default the full image is used.
+void GUIImage::setSourceRect(const core::rect<s32>& sourceRect)
+{
+	SourceRect = sourceRect;
+}
+
+//! Returns the customized source rectangle of the image to be used.
+core::rect<s32> GUIImage::getSourceRect() const
+{
+	return SourceRect;
+}
+
+//! Restrict target drawing-area.
+void GUIImage::setDrawBounds(const core::rect<f32>& drawBoundUVs)
+{
+	DrawBounds = drawBoundUVs;
+	DrawBounds.UpperLeftCorner.X = core::clamp(DrawBounds.UpperLeftCorner.X, 0.f, 1.f);
+	DrawBounds.UpperLeftCorner.Y = core::clamp(DrawBounds.UpperLeftCorner.Y, 0.f, 1.f);
+	DrawBounds.LowerRightCorner.X = core::clamp(DrawBounds.LowerRightCorner.X, 0.f, 1.f);
+	DrawBounds.LowerRightCorner.X = core::clamp(DrawBounds.LowerRightCorner.X, 0.f, 1.f);
+	if ( DrawBounds.UpperLeftCorner.X > DrawBounds.LowerRightCorner.X )
+		DrawBounds.UpperLeftCorner.X = DrawBounds.LowerRightCorner.X;
+	if ( DrawBounds.UpperLeftCorner.Y > DrawBounds.LowerRightCorner.Y )
+		DrawBounds.UpperLeftCorner.Y = DrawBounds.LowerRightCorner.Y;
+}
+
+//! Get target drawing-area restrictions.
+core::rect<f32> GUIImage::getDrawBounds() const
+{
+	return DrawBounds;
+}
+
+//! Writes attributes of the element.
+void GUIImage::serializeAttributes(io::IAttributes* out, io::SAttributeReadWriteOptions* options=0) const
+{
+	IGUIImage::serializeAttributes(out,options);
+
+	out->addTexture	("Texture", Texture);
+	out->addBool	("UseAlphaChannel", UseAlphaChannel);
+	out->addColor	("Color", Color);
+	out->addBool	("ScaleImage", ScaleImage);
+	out->addRect 	("SourceRect", SourceRect);
+	out->addFloat   ("DrawBoundsX1", DrawBounds.UpperLeftCorner.X);
+	out->addFloat   ("DrawBoundsY1", DrawBounds.UpperLeftCorner.Y);
+	out->addFloat   ("DrawBoundsX2", DrawBounds.LowerRightCorner.X);
+	out->addFloat   ("DrawBoundsY2", DrawBounds.LowerRightCorner.Y);
+	out->addBool    ("DrawBackground", DrawBackground);
```

Comment:
We don't use any of this, so these methods should be removed or have empty body.